packages/dds/matrix/src/matrix.ts

```diff
@@ -521,20 +524,31 @@ export class SharedMatrix<T = any>
 			SnapshotPath.cols,
 			this.cols.summarize(this.runtime, this.handle, serializer),
 		);
+		const cellsSnapshot = this.cells.snapshot();
+		const props: Record<string, number> = {
+			csSize: cellsSnapshot.length, // cellsSnapshotSize
+			rc: this.rowCount, // rowCount
+			cc: this.colCount, // colCount
+		};
 		const artifactsToSummarize = [
-			this.cells.snapshot(),
+			cellsSnapshot,
 			this.pending.snapshot(),
 			this.setCellLwwToFwwPolicySwitchOpSeqNumber,
 		];
 
 		// Only need to store it in the snapshot if we have switched the policy already.
 		if (this.setCellLwwToFwwPolicySwitchOpSeqNumber > -1) {
-			artifactsToSummarize.push(this.cellLastWriteTracker.snapshot());
+			const trackerMatrixSnapshot = this.cellLastWriteTracker.snapshot();
+			artifactsToSummarize.push(trackerMatrixSnapshot);
+			props.tmsSize = trackerMatrixSnapshot.length; // trackerMatrixSnapshotSize
 		}
 		builder.addBlob(
 			SnapshotPath.cells,
 			serializer.stringify(artifactsToSummarize, this.handle),
 		);
+		const stats =
+			telemetryContext?.get("fluid:SharedMatrix", "details")?.toString().concat("_") ?? "";
```

Comment:
If I got it right, this joins multiple stringified JSON payloads with "_" as a separator, right?
I'm not sure this is possible to parse back - JSON payload could have same "_" symbols, it's impossible to differentiate between the two types of symbols.